Multi-site implementation - dynamic config approach (#391)
SalesforceCommerceCloud/pwa-kit: packages/pwa/README.md

// config/default.js
module.exports = {
    url: {
        locale: 'path|query_param|none',
        site: 'path|query_param|none',
        showDefaults: false|true
    }
}
```
You can choose how the current locale appears (or doesn’t appear) in the URL by setting `url.locale` to one of the following values:

- `path`: Locale is included in the URL path. Example: `/en-US/women/dress`
- `query_param`: Locale is included as a query parameter. Example: `/women/dress?locale=en-US`
- `none`: Locale isn’t included in the URL. Example: `/women/dress`

`url.showDefaults`: This boolean value dictates whether the default site or locale values are shown in the url. Defaults to: false

By default, a new project is configured to not include the locale and site in the URL path.

## Multi-site Config

By default, the Retail React App is configured to a single locale, single site project.
However, it can be extended to run multiple sites in one single code base. 

Follow these steps to set up your project to support multi-site, multi-locale 
1. Set your url config
- Customise your site and locale in the url 
- Set your url showDefault to true if you want to keep your default values in the url

> 2. Provide alias's for your sites. These will be used in place of your site id when generating paths throughout the application.

Reviewer: I think we should probably move this step to be last and mark it as optional. Since it's not really required that we set this up to get your multi-site up and running. It would also be nice to have a note affirming that site id's can still be used even if alias's are defined.

Author: I've moved it to the last step and marked it optional. Also, I've added some more explanation that the site id can work as a means to determine site and locale and provide an example scenario [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/391/files#diff-374c6fe4a4268dcf10f0e9a3fcffe612bc7e6bf737740599ed8ab8878b00fd34R67)

You can customize how storefront URLs are formatted in your application's configuration file.

```js
// config/default.js
module.exports = {
    url: {
        locale: 'path|query_param|none',
        site: 'path|query_param|none',
        showDefaults: false|true
    }
}
```
You can choose how the current locale appears (or doesn’t appear) in the URL by setting `url.locale` to one of the following values:

- `path`: Locale is included in the URL path. Example: `/en-US/women/dress`
- `query_param`: Locale is included as a query parameter. Example: `/women/dress?locale=en-US`
- `none`: Locale isn’t included in the URL. Example: `/women/dress`

`url.showDefaults`: This boolean value dictates whether the default site or locale values are shown in the url. Defaults to: false

By default, a new project is configured to not include the locale and site in the URL path.

## Multi-site configuration

By default, the Retail React App is configured to a single locale, single site project.
However, it can be extended to run multiple sites in one single code base. 

Follow these steps to set up your project to support multi-site, multi-locale 
1. Set your url config

> - Customise your site and locale in the url 

Reviewer: ```suggestion
- Customize your site and locale in the url 
```

> - Set your url showDefault to true if you want to keep your default values in the url

Reviewer: ```suggestion
- Set your url `showDefault` to true if you want to keep your default values in the url
```

2. Provide the sites for your app. Each site includes site id, and its localization configuration.

>    You can also provide alias for your locale. They will be used in place of your locale id when generating paths across the app

Reviewer: ```suggestion
   You can also provide an alias for your locale that will be used in place of your locale id when generating paths across the app.
```

> 3. [Optional] Provide alias's for your sites. These will be used in place of your site id when generating paths throughout the application.

Reviewer: ```suggestion
3. [Optional] Provide aliases for your sites. These will be used in place of your site id when generating paths throughout the application.
```

If no alias is defined for a site, the id will be used to generate paths across application. 

>    *Note*:  Even if alias is defined, the URL that has id instead of the alias would still work as a mean to determine the site and locale for the app, 
> but be mindful that the generated paths will use alias. 
>    This is helpful in some cases. For example: At some time, your application URLs have id, but then you switch to use alias.
> If a user bookmarks a link that has id in the path before you deploy your change, the old bookmark link still works when you use it to your site.   

Reviewer: No need to get too in depth about this reasoning.

```suggestion
   *Note*:  URLs constructed using canonical site and locale ids are still valid URLs even when aliases are used. 
```
